docs/spec.md

```diff
@@ -2244,26 +2262,29 @@ the XLA compiler. In the future, we are planning to unify this metadata
 Performs element-wise division of dividend `lhs` and divisor `rhs` tensors and
 produces a `result` tensor. Depending on the element type, does the following:
 
-* For integers: integer division.
+* For integers: integer division which produces the algebraic quotient with any
+  fractional part discarded.
 * For floats: `division` from IEEE-754.
 * For complex numbers: complex division.
+* For quantized types:
+  * `dequantize_op_quantize(divide, lhs, rhs, type(result))`.
 
 #### Inputs
 
-| Label | Name  | Type                                              | Constraints |
-|-------|-------|---------------------------------------------------|-------------|
-| (I1)  | `lhs` | tensor of integer, floating-point or complex type | (C1)        |
-| (I2)  | `rhs` | tensor of integer, floating-point or complex type | (C1)        |
+| Label | Name  | Type                                                                              | Constraints |
+|-------|-------|-----------------------------------------------------------------------------------|-------------|
+| (I1)  | `lhs` | tensor of integer, floating-point, or complex type or per-tensor quantized tensor | (C1)        |
+| (I2)  | `rhs` | tensor of integer, floating-point, or complex type or per-tensor quantized tensor | (C1)        |
 
 #### Outputs
 
-| Name     | Type                                              | Constraints |
-|----------|---------------------------------------------------|-------------|
-| `result` | tensor of integer, floating-point or complex type | (C1)        |
+| Name     | Type                                                                              | Constraints |
+|----------|-----------------------------------------------------------------------------------|-------------|
+| `result` | tensor of integer, floating-point, or complex type or per-tensor quantized tensor | (C1)        |
 
 #### Constraints
 
-* (C1) `type(lhs) = type(rhs) = type(result)`.
+* (C1) `baseline_type(lhs) = baseline_type(rhs) = baseline_type(result)`.
```

Comment:
Thanks @wecing for the proposal. 

>> I think option 1 would be slightly better, since a more flexible IR is easier to manipulate.

I agree with that , but , IMHO, this comes at the cost of making the op complex by fusing a re-quantize in the semantics of divide. Can you please elaborate a bit of the benefit on such a fused version of `divide` op? Maybe it will lead to less pattern matching by the downstream consumers of StableHLO. IMO, such overhead is not too much, preserving the  generality of op plus the associated separation of concerns  is something worth considering. 
   
On a related note: Is divide is the only op which needs this change or there are other candidates (maybe `rsqrt`) which might need this similar  treatment?
 
cc @burmako

Reply:
> this comes at the cost of making the op complex

> such overhead is not too much

Agreed.

> Is divide is the only op which needs this change

`stablehlo.add` also has this problem. For convolution followed by bias add, we may want convolution to return `qi32` and bias add be `(qi32, qi32) -> qi8`.

But since this usage is only applicable to a few ops, I think preferring simplicity in IR design and keeping the current proposal is fine.

Reply:
>> stablehlo.add also has this problem. For convolution followed by bias add, we may want convolution to return qi32 and bias add be (qi32, qi32) -> qi8.

Thanks for the information. 

>> But since this usage is only applicable to a few ops, I think preferring simplicity in IR design and keeping the current proposal is fine.
Thanks for the feedback! Agreed.